nltools/data/brain_data.py

```diff
@@ -1229,36 +1267,67 @@ def apply_mask(self, mask):
         if (len(masked.shape()) > 1) & (masked.shape()[0] == 1):
             masked.data = masked.data.flatten()
 
-    def extract_roi(self, mask, method='mean'):
+    def extract_roi(self, mask, metric='mean', n_components=None):
         """ Extract activity from mask
 
         Args:
-            mask: (nifiti) nibabel mask can be binary or numbered for
+            mask: (nifti) nibabel mask can be binary or numbered for
                   different rois
-            method: type of extraction method (default=mean)
+            metric: type of extraction method ['mean', 'median', 'pca'], (default=mean)
                     NOTE: Only mean currently works!
+            n_components: if metric='pca', number of components to return (takes any input into sklearn.Decomposition.PCA)
 
         Returns:
             out: mean within each ROI across images
 
         """
         mask = check_brain_data(mask)
         ma = mask.copy()
-
-        if method != 'mean':
-            raise ValueError('Only mean is currently implemented.')
+            
+        if metric not in metrics:
```

Comment:
`metrics` is undefined in the function body

Reply:
already fixed.